developer_manual/app_publishing_maintenance/release_process.rst

```diff
@@ -0,0 +1,127 @@
+* Any other change that makes the app incompatible with a previously compatible environment (breaking change)
+
+.. tip:: Example: the app is at version 3.7.2. The next major version will be 4.0.0.
+
+Change log
+~~~~~~~~~~
+
+If the app keeps a change log (e.g. CHANGELOG.md in the project root) it's time to update it with all **added**, **changed** and **fixed** items.
+
+.. tip:: Use ``git log v3.7.1..HEAD --pretty=oneline | grep -v tx-robot | grep -v "Merge " | grep -v "Bump "`` on the target release branch to get a compact list of all changes since the v3.7.1 release. Adjust this to your previous release version number.
+
+Pre-releases
+~~~~~~~~~~~~
+
+Optionally maintainers may chose to pre-release their apps on the app store for early adopters and testers. The precise periodization is up to the maintainer. Common periods include **alpha**, **beta** and **rc** (release candidate) epochs.
+
+The release process is identical to the one of a final release, only the version number has a suffix.
+
+.. tip:: Example: the app will be released as version 3.7.2. Therefore there might be a **v3.7.2-alpha.1**, **v3.7.2-beta.1**, **v3.7.2-rc.1**, **v3.7.2-rc.1** and the final **v3.7.2**.
+
+
+The release
+-----------
+
+From an abstract point of view the main part of doing a release is transforming the source code into an installable software component.
+
+This part is typically scripted and highly depends on the type of app. The following list is incomplete but should give a rough idea of what steps a release script should contain:
+
+* Install all :ref:`dependencies <app-dependencies>`
+    * Run ``composer i --no-dev`` if the app uses :ref:`Composer <app-composer>`
+    * Run ``npm ci`` if the app uses :ref:`npm <app-npm>`
+* Build compiled artifacts
+    * :ref:`Build production scripts for the front-end <app-npm-build>`
+    * Run any code generation (e.g. through a :ref:`Composer script <app-composer>`)
+* Remove development files
+    * Remove any kind of configuration files (composer.*, package.json, package-lock.json, .babelrc, and so on) that are not required in production
+    * Remove source code that is not required in production, e.g. JavaScript that is compiled into a bundle
+    * Remove tests
+* :ref:`Sign the release files <app-code-signing>` to generate an appinfo/signature.json
+* Package the rest into a tarball
+* Upload the tarball for distribution, e.g. as a Github release artifact or a dedicated download server
+* :ref:`Publish <app-store-publishing>` on the app store
+* Tag the release in Git and push your local changes, if any
+
+After the release
+-----------------
+
+Branch off
+~~~~~~~~~~
+
+If the maintainer of the app keeps stable branches to which bug fixes are backported, any major or minor release requires a branching off the current main branch.
+
+.. tip:: Example: the app is released as version 3.7.0. The new latest stable branch will be ``stable3.7``.
```

Comment:
Could also explain the stable23 schema we use in many other apps?

Reply:
> This should (correct me if something bothers you) be free of any subjective preferences in release processes, e.g. with Nextcloud support ranges, versioning, automation, tooling and so on.

I'd rather just drop my *tip* than add all possible release philosophies.